syntax = "proto3";

enum RequestCode {
    USER_LOGIN = 0;
    REGISTER = 1;
    AES_DISTRIBUTION = 2;
    AUTO_CREATE_ROOM = 3;
    MANUAL_CREATE_ROOM = 4;
    SEARCH_ROOM = 5;
}

enum ResponseCode {
    LOGIN_OK = 0;
    REGISTER_OK = 1;
    RSA_DISTRIBUTION = 2;
    AES_VERIFY_OK = 3;
    JOIN_ROOM_OK = 4;
    START_GAME = 5;
    SEARCH_ROOM_OK = 6;
    FAILED = 7;
}

message Information {
    string user_name = 1;
    string room_name = 2;
    string data1 = 3;
    string data2 = 4;
    string data3 = 5;
    RequestCode reqcode = 6;
    ResponseCode rescode = 7;
}